### ISSUE TITLE
feature: size limit on caching filesystem

### ISSUE THREAD

**Issue by majidaldo** [intake/filesystem_spec#510]
It would be nice to set a limit on the size of the caching filesystems. This would truly enable 'set and forget' kind of work instead of having to manually go into the cache location to clear things out.

**Comment by martindurant** [comment]
I can imagine such a feature, and wouldn't be too hard to implement. You would clear old files by timestamp, oldest first. On simplecache, it would mean having to list files occasionally; on other ones to keep the metadata up to date.

**Comment by JulioHC00** [comment]
Hi, is there any solution available now for limiting the cache size?

**Comment by RoelantStegmann** [comment]
I agree that this would be very nice. I would like to add that in simple caching (so not block caching), it should also not even attempt to cache a file that is in itself larger than the cache size.
